File: Civitas-Java/Civitas/src/main/java/civitas/Casilla.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/**
 * @author elena
 * @date 20/09/2019
 */

package civitas;

import java.util.ArrayList;

public class Casilla {
    private String nombre; 
    
    Casilla(String nombre){
        this.nombre = nombre;
    }
    
    public String getNombre(){ return nombre; }
    
    void informe(int iactual, ArrayList<Jugador> todos){
        Diario diario = Diario.getInstance();
        diario.ocurreEvento("El jugador " + todos.get(iactual).getNombre() + " ha caido en la casilla " + toString());
    }
    
    public boolean jugadorCorrecto(int iactual, ArrayList<Jugador> todos){ return iactual < todos.size(); }
    
    void recibeJugador(int iactual, ArrayList<Jugador> todos){        
        informe(iactual, todos);
    }   
    
    @Override
    public String toString(){
        return "Nombre: " + this.nombre +
               "\nTipo Casilla: DESCANSO";
    }
}

File: Civitas-Ruby/lib/casilla.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative "diario"
require_relative "mazo_sorpresas"
require_relative "titulo_propiedad"
require_relative "sorpresa"
require_relative "tipo_casilla"
require_relative "tipo_sorpresa"
require_relative "jugador"

module Civitas
  class Casilla
    attr_reader :nombre
    public :nombre
    
    def initialize(nombre)
      @nombre = nombre
    end
           
    def informe(actual, todos)
      jugador = todos[actual]
      evento = "El jugador " + "#{jugador.nombre} ha caido en la casilla  #{to_s}"
      Diario.instance.ocurre_evento(evento)
    end
    
    def recibejugador(actual, todos)  
        informe(actual, todos)      
    end
        
    public 
    
    def jugadorcorrecto(actual, todos)
      actual < todos.lenght
    end
    
    def to_s
        '\nNombre: ' + @nombre.to_s +
        '\nTipo Casilla: DESCANSO'      
    end
    
    public :nombre, :titulo
  end
end
